src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionFactory.java

```diff
@@ -108,6 +109,15 @@ public static MySqlConnectionFactory from(MySqlConnectionConfiguration configura
                 });
         }));
     }
+    
+    @Override
+    public Mono<Void> close() {
+    	return Mono.defer(() -> {
+    		this.client = null;
```

Comment:
Hello @svats0001,

I hope you're doing well. Unfortunately, I'm unable to accept this change at the moment, as it appears not to be thread-safe and there seems to be no handling for a null client. Would it be possible for us to postpone this change and perhaps discuss potential solutions to address these concerns?

Thank you for your understanding.

Reply:
Hi @jchrys,

Yep that makes sense. I'll make some changes and create another PR at a later date. I'll have a further read about thread safety but would a synchronized block on the client object work? I can add handling for the null client in the close method, is the create() method in MySqlConnectionFactory the only other place that I need to handle the null client?